internal/controller/nodemaintenance_controller.go

```diff
@@ -600,15 +627,38 @@ func (p ReadyConditionChangedPredicate) Update(e event.TypedUpdateEvent[client.O
 		return false
 	}
 
-	oldRCR := k8sutils.GetReadyConditionReason(oldO)
-	newRCR := k8sutils.GetReadyConditionReason(newO)
+	condChanged := false
+
+	oldConds := make(map[string]metav1.Condition)
```

Comment:
nit: why can't we sort the condition slice and then simply compare with reflect.DeepEqual?

Reply:
great idea :) will do it ofc.

Reply:
@ykulazhenkov PTAL, LMK if thats what you had in mind.